File: config_ini.R
# install.packages("configr")
library(configr)


# INI
config <- read.config("config.ini")

print(config)

cat(config$modul1$zmienna_txt)
cat(config$modul1$zmienna_num)

# klasa liczby
cat(class(config$modul1$zmienna_num))
cat(class(config$modul1$zmienna_flo))


cat(config$modul2$zmienna_arr_txt)
cat(config$modul2$zmienna_arr_num)

# uwaga - wunik jest stringiem a nie listą!
cat(config$modul2$zmienna_arr_num[2])

File: config_ini.py
import configparser
import pprint

pp = pprint.PrettyPrinter(indent=2)


config = configparser.ConfigParser()
config.read('config.ini')

pp.pprint(config)

print(config.get('modul1', 'zmienna_txt'))
print(config.get('modul1', 'zmienna_num'))

print(type(config['modul1']['zmienna_num']))
print(type(config['modul1']['zmienna_flo']))

print(config['modul2']['zmienna_arr_txt'])
print(config['modul2']['zmienna_arr_txt'][1])
print(config['modul2']['zmienna_arr_num'][1])
